### DISCUSSION TITLE
Nuitka to compile binaries for linux and Mac

### DISCUSSION THREAD

**Discussion by Kurdakov** [ThioJoe/YT-Spammer-Purge#258] [Ideas]
after installing Nuitka I managed to compile self contained linux binary

python -m nuitka --lto=yes --follow-imports --static-libpython=yes --onefile  --include-data-dir=/path/to/assets=assets  YouTubeSpammerPurge.py 

(the question is though if it will run on different linuxes - but in principle, it should, the same concerns Macs - so there is need to test such binaries) 

on windows it is also possible to include icon, for example

python -m nuitka --lto=yes --follow-imports --onefile   --mingw64 --include-data-dir=C:/YouTube-Spammer-Purge/assets=assets --standalone --windows-icon-from-ico=C:/YouTube-Spammer-Purge/icon.png YouTubeSpammerPurge.py

but the result is less than impressive - too many files (dlls) and also one needs manually copy confusables directory from site-packages to dist directory of compiled exe 

so if someone has Mac to compile binary and then there are others who will test on other Macs, maybe it's possible to make binaries for Mac and then possibly even for linux (as they seem to be more compact than nuitka output on windows - just one bin file)

p.s. I did not notice any significant speedup due to compilation, but loading seems a little bit more quick, than with stock python

btw cancel_button.png executeButton.png in assets - are they needed? seems not, so should be deleted from repository

**Comment by rbeer** [comment]
At least executeButton.png is used in the gui window for setting filter characters and words.

https://github.com/ThioJoe/YT-Spammer-Purge/blob/8bc7a34a5fd182c2cb58aacb8788043608fb99ef/gui.py#L242

**Reply by Kurdakov** [reply]
found that pycompiler also is multiplatform (linux and mac and also has options for signing - for nuitka a separate signing step will be needed) and given that no speed improvements are observed no need for nuitka (except possibly to solve openssl incompability issue, as nuitka makes all needed dlls local - but then, there will be need to package all these dlls along with exe, which is inconvenient)
